src/BotwinExtensions.cs

```diff
@@ -4,36 +4,32 @@ namespace Botwin
     using System.IO;
     using System.Linq;
     using System.Reflection;
-    using System.Threading.Tasks;
     using FluentValidation;
     using Microsoft.AspNetCore.Builder;
     using Microsoft.AspNetCore.Http;
     using Microsoft.AspNetCore.Routing;
     using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
 
     public static class BotwinExtensions
     {
-        /// <summary>
-        /// Adds Botwin to the specified <see cref="IApplicationBuilder"/>.
-        /// </summary>
-        /// <param name="builder">The <see cref="IApplicationBuilder"/> to configure.</param>
-        /// <returns>A reference to this instance after the operation has completed.</returns>
-        public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder)
-        {
-            return UseBotwin(builder, null);
-        }
-
         /// <summary>
         /// Adds Botwin to the specified <see cref="IApplicationBuilder"/>.
         /// </summary>
         /// <param name="builder">The <see cref="IApplicationBuilder"/> to configure.</param>
         /// <param name="options">A <see cref="BotwinOptions"/> instance.</param>
         /// <returns>A reference to this instance after the operation has completed.</returns>
-        public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, BotwinOptions options)
+        public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, BotwinOptions options = null)
         {
-            ApplyGlobalBeforeHook(builder, options);
+            var loggerFactory = builder.ApplicationServices.GetService<ILoggerFactory>();
```

Comment:
`loggerFactory` will never be `null` here, so no need to have any of the following null-coalescing calls.